packages/mermaid/src/diagrams/class/classDb.ts

```diff
@@ -217,84 +220,93 @@ export const cleanupLabel = function (label: string) {
 /**
  * Called by parser when assigning cssClass to a class
  *
- * @param ids - Comma separated list of ids
- * @param className - Class to add
+ * @param classNames - Comma separated list of ids
+ * @param cssClass - Class to add
  */
-export const setCssClass = function (ids: string, className: string) {
-  ids.split(',').forEach(function (_id) {
-    let id = _id;
-    if (_id[0].match(/\d/)) {
-      id = MERMAID_DOM_ID_PREFIX + id;
+export const setCssClass = function (classNames: string, cssClass: string) {
+  classNames.split(',').forEach(function (_className) {
+    let className = _className;
+    addClass(className);
+
+    if (_className[0].match(/\d/)) {
+      className = MERMAID_DOM_ID_PREFIX + className;
     }
-    if (classes[id] !== undefined) {
-      classes[id].cssClasses.push(className);
+    if (classes[className] !== undefined) {
+      classes[className].cssClasses.push(cssClass);
     }
   });
 };
 
 /**
  * Called by parser when a tooltip is found, e.g. a clickable element.
  *
- * @param ids - Comma separated list of ids
+ * @param classNames - Comma separated list of ids
  * @param tooltip - Tooltip to add
  */
-const setTooltip = function (ids: string, tooltip?: string) {
-  ids.split(',').forEach(function (id) {
+const setTooltip = function (classNames: string, tooltip?: string) {
+  classNames.split(',').forEach(function (className) {
     if (tooltip !== undefined) {
-      classes[id].tooltip = sanitizeText(tooltip);
+      addClass(className);
+      classes[className].tooltip = sanitizeText(tooltip);
     }
   });
 };
 
-export const getTooltip = function (id: string, namespace?: string) {
+export const getTooltip = function (className: string, namespace?: string) {
   if (namespace) {
-    return namespaces[namespace].classes[id].tooltip;
+    return namespaces[namespace].classes[className].tooltip;
   }
 
-  return classes[id].tooltip;
+  return classes[className].tooltip;
 };
 
 /**
  * Called by parser when a link is found. Adds the URL to the vertex data.
  *
- * @param ids - Comma separated list of ids
+ * @param classNames - Comma separated list of class ids
  * @param linkStr - URL to create a link for
  * @param target - Target of the link, _blank by default as originally defined in the svgDraw.js file
  */
-export const setLink = function (ids: string, linkStr: string, target: string) {
+export const setLink = function (classNames: string, linkStr: string, target: string) {
   const config = getConfig();
-  ids.split(',').forEach(function (_id) {
-    let id = _id;
-    if (_id[0].match(/\d/)) {
-      id = MERMAID_DOM_ID_PREFIX + id;
+  classNames.split(',').forEach(function (_className) {
+    let className = _className;
+    if (_className[0].match(/\d/)) {
+      className = MERMAID_DOM_ID_PREFIX + className;
     }
-    if (classes[id] !== undefined) {
-      classes[id].link = utils.formatUrl(linkStr, config);
+    addClass(className);
+    if (classes[className] !== undefined) {
```

Comment:
This `if()` statement will always be `true`, since we always call `addClass(className)` before it.

Reply:
Hmmmmm, that's a good point. We could do something like:

```ts
function addClass(className: string): asserts classes[className] !== undefined {
```

See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

The issue is that calling `addClass('Array~string~')` creates `classes['Array']`, not `classes['Array~string~']`, so we'd probably need a new function like `addClassID`.

Reply:
`addClass('Array~string~')` is SUPPOSED to create classes['Array'].  The type is a property on the class object, and not part of the id.

At least, that is the way it is now.  PR https://github.com/mermaid-js/mermaid/pull/5223 will change this so the generic type will be appended to the id. `addClass(Array~string~)` will create `classes['Array-string']`